proposals/browser_test_api.md

```diff
@@ -0,0 +1,173 @@
+# Proposal: browser.test API
+
+**Summary**
+
+Enable testing using the `browser.test` runner and assertion API.
+
+**Document Metadata**
+
+**Authors:** @kiaraarose and @zombie
+
+**Sponsoring Browser:** Mozilla Firefox
+
+**Created:** 2025-07-17
+
+**Related Issues:** web-platform-tests/rfcs/pull/219
+
+## Motivation
+
+### Objective
+
+This API provides utilities for writing and running tests for WebExtensions.
+It includes assertion functions and a test harness to facilitate structured
+and reliable testing.
+
+#### Use Cases
+
+The primary use case is testing WebExtensions implementations using
+Web Platform Tests.  Secondary use case is enabling developers to
+write and run a cross-browser test suite for their own extensions.
+
+### Known Consumers
+
+All browsers already have this API namespace, and the goal of specifying it
+here is to align existing implementations around common behaviors.
```

Comment:
this is doing more than just specifying existing implementations / common behaviors; lots of this doesn't work in Chromium today.  That's okay (we've acknowledged there's work to be done here and this won't work out of the box), but I think it's worth highlighting that the intention of this proposal *is* to introduce behavioral changes in some browsers.

Reply:
Good point, I added a more explicit statement that this will require implementation changes.